rotkehlchen/db/dbhandler.py

```diff
@@ -1286,49 +1270,51 @@ def get_yearn_vaults_events(
-            'pnl_amount, '
-            'pnl_usd_value, '
-            'block_number, '
-            'timestamp, '
-            'tx_hash, '
-            'log_index '
-            'from yearn_vaults_events WHERE address=? AND (from_asset=? OR from_asset=?);',
+            'SELECT * from yearn_vaults_events WHERE address=? '
+            'AND (from_asset=? OR from_asset=?);',
             (address, vault.underlying_token.identifier, vault.token.identifier),
         )
         events = []
         for result in query:
-            realized_pnl = None
-            if result[7] is not None:
-                realized_pnl = Balance(amount=FVal(result[7]), usd_value=FVal(result[8]))
-            events.append(YearnVaultEvent(
-                event_type=result[0],
-                from_asset=Asset(result[1]),
-                from_value=Balance(amount=FVal(result[2]), usd_value=FVal(result[3])),
-                to_asset=Asset(result[4]),
-                to_value=Balance(amount=FVal(result[5]), usd_value=FVal(result[6])),
-                realized_pnl=realized_pnl,
-                block_number=int(result[9]),
-                timestamp=Timestamp(int(result[10])),
-                tx_hash=result[11],
-                log_index=result[12],
-            ))
+            try:
+                events.append(YearnVaultEvent.deserialize_from_db(result))
+            except DeserializationError as e:
+                msg = f'Failed to read yearn vault event from database {result}'
+                self.msg_aggregator.add_warning(msg)
+                log.error(msg, error=str(e))
         return events
 
-    def delete_yearn_vaults_data(self) -> None:
-        """Delete all historical aave event data"""
+    def get_all_yearn_vaults_v2_events(self, address: ChecksumEthAddress) -> List[YearnVaultEvent]:
         cursor = self.conn.cursor()
-        cursor.execute('DELETE FROM yearn_vaults_events;')
-        cursor.execute(f'DELETE FROM used_query_ranges WHERE name LIKE "{YEARN_VAULTS_PREFIX}%";')
+        query = cursor.execute(
+            'SELECT * from yearn_vaults_events WHERE address=? AND version=2;',
+            (address, ),
+        )
+        events = []
+        for result in query:
+            try:
+                events.append(YearnVaultEvent.deserialize_from_db(result))
+            except DeserializationError as e:
+                msg = f'Failed to read yearn vault event from database {result}'
+                self.msg_aggregator.add_warning(msg)
+                log.error(msg, error=str(e))
+        return events
+
+    def delete_yearn_vaults_data(self, version: int = 1) -> None:
+        """Delete all historical yearn vault events data"""
+        if version not in (1, 2):
+            log.error(f'Called delete yearn vault data with non valid version {version}')
+            return None
+        if version == 1:
+            prefix = YEARN_VAULTS_PREFIX
+        elif version == 2:
+            prefix = YEARN_VAULTS_V2_PREFIX
```

Comment:
no need for 2nd check right now. 
```suggestion
        prefix = YEARN_VAULTS_PREFIX
        if version == 2:
            prefix = YEARN_VAULTS_V2_PREFIX
```